File: 2022W06/B0neh3ad/boj_3111.py
a = input()
t = input()
an = len(a)
s = 0
e = len(t)
while True:
    idx = t[s:e].find(a) + s
    if idx == s-1:
        break
    t = t[:idx] + t[idx+an:]
    s = max(idx-an, 0)
    e = max(idx+an, e-an)
    
    idx = t[s:e].rfind(a) + s
    if idx == s-1:
        break
    t = t[:idx] + t[idx+an:]
    s = min(idx-an, s)
    e = min(idx+an, len(t))

print(t, end='')

File: 2022W06/B0neh3ad/boj_3111.cpp
#include <bits/stdc++.h>
using namespace std;

int main(void){
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    string a, t;
    cin >> a >> t;

    int s = 0, e = t.size()-1;
    int an = a.size();
    bool flag = false;
    while(true){
        flag = false;
        for(int i=s; i<t.size(); ++i){
            if(t.substr(i, an) == a){
                t.erase(i, an);
                s = max(i - an, 0);
                e = max(e, i + an);
                flag = true;
                break;
            }
        }
        if(!flag){ break; }
        if(e > t.size()-1){ e = t.size()-1; }

        flag = false;
        for(int i=e-an+1; i>=0; --i){
            if(t.substr(i, an) == a){
                t.erase(i, an);
                e = min(i + an, int(t.size())-1);
                s = min(s, i - an);
                flag = true;
                break;
            }
        }
        if(!flag){ break; }
    }

    cout << t;
    return 0;
}